src/moduleInstance.js

```diff
@@ -222,36 +233,42 @@ export function makeModuleInstance(
               notify(updater);
             }
           }
+          if (exportAlls.includes(specifier)) {
+            // Make all these imports candidates.
+            for (const [importName, notify] of entries(modNotifiers)) {
+              if (candidateAll[importName] === undefined) {
+                candidateAll[importName] = notify;
+              } else {
+                // Already a candidate: remove ambiguity.
+                candidateAll[importName] = false;
+              }
+            }
+          }
         });
       ps.push(p);
     }
     await Promise.all(ps);
-    // Execute the export * from 'mod' clauses in order.
-    for (const specifier of exportAlls) {
-      const moduleId = linkageRecord.moduleIds[specifier];
-      const instance = importNS.get(moduleId);
-      // eslint-disable-next-line no-await-in-loop
-      await instance.initialize().then(() => {
-        const { notifiers: modNotifiers } = instance;
-        for (const [importName, notify] of Object.entries(modNotifiers)) {
-          if (importName !== 'default' && !notifiers[importName]) {
-            notifiers[importName] = notify;
-
-            // exported live binding state
-            let value;
-            notify(v => (value = v));
-            defProp(moduleNS, importName, {
-              get() {
-                return value;
-              },
-              set: undefined,
-              enumerable: true,
-              configurable: false,
-            });
-          }
-        }
-      });
+
+    for (const [importName, notify] of Object.entries(candidateAll)) {
+      if (!notifiers[importName] && notify !== false) {
+        notifiers[importName] = notify;
+
+        // exported live binding state
+        let value;
+        notify(v => (value = v));
+        moduleNSProps[importName] = {
+          get() {
+            return value;
+          },
+          set: undefined,
+          enumerable: true,
+          configurable: false,
+        };
+      }
     }
+
+    // Sort the module namespace.  Some es262 tests rely on this behaviour.
```

Comment:
Maybe we should say "as per specifications" since we're not fixing to pass the test, but to conform to specifications and avoid runtime failures modules relying to that behavior.